client/src/app/_components/groups/group-post-input/index.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable react/no-children-prop */
+import React, { type SetStateAction, useState } from 'react';
+import { TextInput } from '../../ui/text-input';
+import { TextArea } from '../../ui/text-area';
+import { BasicButton } from '../../ui/basic-button';
+import { BasicModal } from '../../ui/basic-modal';
+import { useWallet } from '~/providers/walletprovider';
+import { api } from '~/trpc/react';
+import { DateTime } from 'luxon';
+
+type GroupPostInputProps = {
+	groupId: string;
+	onPostSubmit: () => void;
+};
+
+const GroupPostInput = ({ groupId, onPostSubmit }: GroupPostInputProps) => {
+	const [postOpen, setPostOpen] = useState(false);
+	const [title, setTitle] = useState('');
+	const [content, setContent] = useState('');
+	const [isLoading, setIsLoading] = useState(false);
+	const { isConnected, walletAddress } = useWallet();
+
+	const postToIPFS = api.PostToIPFS.postMessage.useMutation();
+	const postToFirebase = api.PostToFirebase.postToCollection.useMutation();
+
+	const hidePostInput = () => {
+		setPostOpen(false);
+	};
+
+	const showPostInput = () => {
+		setPostOpen(true);
+	};
+
+	const handleTitle = (e: { target: { value: SetStateAction<string> } }) => {
+		setTitle(e.target.value);
+	};
+
+	const handleContent = (e: { target: { value: SetStateAction<string> } }) => {
+		setContent(e.target.value);
+	};
+
+	const closeAndClear = () => {
+		hidePostInput();
+		setTitle('');
+		setContent('');
+		onPostSubmit();
+	};
```

Comment:
This won't be needed when you merge, this is already being handled by the validation library